### ISSUE TITLE
PyZMQ with tornado 6.0 raises the wrong warning.

### ISSUE THREAD

**Issue by Zsailer** [zeromq/pyzmq#1310]
PyZMQ raises the following warning when users upgrade to tornado>=6.x:

```bash
Users/klay6683/miniconda3/envs/py37/lib/python3.7/site-packages/notebook/services/kernels/kernelmanager.py:19: VisibleDeprecationWarning: zmq.eventloop.minitornado is deprecated in pyzmq 14.0 and will be removed.
    Install tornado itself to use zmq with the tornado IOLoop.
```
_Originally posted by @michaelaye in https://github.com/jupyter/notebook/issues/4549#issuecomment-509817150_

This happens because pyzmq attempts to import a module in tornado that no longer exists (specifically, [this import chunk](https://github.com/zeromq/pyzmq/blob/ae4895545c5053ea6205f9eef9388ea65199447b/zmq/eventloop/zmqstream.py#L40-L49)). When it cannot find this module, it **assumes tornado is not installed**, tries to import a deprecated module inside pyzmq (i.e. minitornado), and triggers this warning. Tornado *is* of course installed—it just no longer contains this specific module—so this warning isn't communicating the correct problem.

To fix, we either pull out the imported function from the deprecated module *or* stop using this function since it doesn't exist in Tornado anymore.

**Comment by minrk** [comment]
This was fixed in 17.1.3 by #1263 and does not affect pyzmq 18 or master.

The block you linked includes the fix only imports minitornado *if it's already been imported*, so a deprecation warning caused by importing minitornado would be thrown before this block executes.

Annotated code:

```python
try:
    from tornado.stack_context import wrap as stack_context_wrap  # raises ImportError on tornado 6
except ImportError:
    if "zmq.eventloop.minitornado" in sys.modules:  # False if minitornado is not already imported, which would mean warning is already shown
        # not executed if minitornado is not already imported
        from .minitornado.stack_context import wrap as stack_context_wrap
    else:
        # this is what will normally happen with tornado 6
        # tornado 5 deprecates stack_context,
        # tornado 6 removes it
        def stack_context_wrap(callback):
            return callback
```

To demonstrate that this is not the case, run:

```bash
python -c 'import zmq.eventloop.zmqstream'
```
 which executed the above block and does not produce a warning.

**Comment by Zsailer** [comment]
Ah, thanks @minrk. I tracked down this block of code but obviously didn't look at this code close enough.

Thank you for following up!
